### ISSUE TITLE
extract translations from url

### ISSUE THREAD

**Issue by callumrollo** [callumrollo/erddaplogs#59]
translations are injected into the urls as an iso3166 code after base_url/erddap e.g the German translation of 

https://erddap.observations.voiceoftheocean.org/erddap/files/documentation.html

is made by adding `de` into the url

https://erddap.observations.voiceoftheocean.org/erddap/de/files/documentation.html

We can extract this with a bit of splitting the url column iin polars:

```python
df = df.with_columns(
    pl.col("url")
    .str.split_exact("/", 3)
    .struct.rename_fields(["blank", "root","first", "rest"])
    .alias("fields")
).unnest("fields")
```

The translation, if present, will be in the column named `first`. In non-translated pages, this part of the url is usually `griddap`, `tabledap`, `files` etc., and is currently naievely used by erddaplogs to classify the request. This will silently fail for translated pages.

We need to handle this. Could be expensive

**Comment by ChrisJohnNOAA** [comment]
Just for context, the way ERDDAP itself handles this is here:
https://github.com/ERDDAP/erddap/blob/main/WEB-INF/classes/gov/noaa/pfel/erddap/Erddap.java#L568-L592

Essentially get first section from the url, and lookup if it's a supported language. If it's not a supported language, then assume it's a protocol.

**Comment by callumrollo** [comment]
My current solution is to take the list of supported langauges from here, as the codes don't seem to exactly match any of the ISO standards  

https://github.com/ERDDAP/erddap/blob/72a2fbee389209c22f051b5ccf49e4a6b4c8d5a8/WEB-INF/classes/gov/noaa/pfel/erddap/util/TranslateMessages.java#L197

Will put in a PR tomorrow

I'm going to make more tests/checks to ensure we're correctly identifying/classifying protocols

**Comment by callumrollo** [comment]
Resolved in #60
